simgrid: 3.21 -> 3.22.2

mpoquet opened the pull request:
###### Motivation for this change
Keep nixpkgs's SimGrid up-to-date.

This commit does the following.
- Update git repository. The SimGrid team uses Framagit more than GitHub now, and intermediate releases (as this one, 3.22.2) are not available on GitHub.
- In checkPhase, explicitly compile test binaries thanks to `make tests`. This is now required as calling `make` without parameter no longer build test binaries.
- Fix broken documentation link.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distribution (Arch + Nix)
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

mpoquet committed:
simgrid: 3.21 -> 3.22.2

- GitHub -> Framagit (SimGrid main repo has moved)
- Test binaries must now be explicitly compiled (`make tests`)
- Fix broken documentation link

JohnAZoidberg commented on pkgs/applications/science/misc/simgrid/default.nix:
```diff
@@ -87,6 +86,7 @@ stdenv.mkDerivation rec {
   checkPhase = ''
     runHook preCheck
 
+    make tests -j $NIX_BUILD_CORES
```
`enableParallelChecking` already does that and it's `true` by default.
So you just need to keep the default `checkPhase`.
You could move

```bash
ctest -j $NIX_BUILD_CORES --output-on-failure -E smpi-replay-multiple
```

to `postCheck`, remove the rest and it'll work.

mpoquet replied:
Unfortunately, the default `checkPhase` fails before calling postCheck.
It calls CMake tests without excluding `smpi-replay-multiple`, which fails.

JohnAZoidberg replied:
`--output-on-failure` is enabled by default in our nixpkgs toolchain.
To disable a test you can add `-DCTEST_CUSTOM_TESTS_IGNORE=smpi-replay-multiple` to the `cmakeFlags`

JohnAZoidberg replied:
Took me a while but I solved it using those two resources:
- https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/build-managers/cmake/setup-hook.sh
- https://gitlab.kitware.com/cmake/community/wikis/doc/ctest/Testing-With-CTest

mpoquet replied:
I tried ([patch](https://pastebin.com/qSJaTT90)) to set `CTEST_CUSTOM_TESTS_IGNORE` (and friends: `CTEST_CUSTOM_MEMCHECK_IGNORE`, `CTEST_CUSTOM_COVERAGE_EXCLUDE`) via `cmakeFlags` but this is ignored by the default `checkPhase`: It runs the test and fails.

I could generate a `CTestCustom.cmake` in the build tree but I'm not convinced this is the way to go. I feel like it would add more complexity than redefining `checkPhase`.

JohnAZoidberg replied:
Yeah okay, sorry. It doesn't work :/

But it doesn't have be be an entirely new `cmake` file, right? Can't you just append a single line to the current one?
If not just leave it as it is right now.

mpoquet replied:
Should be fixed by 00fc5bd08fc, which ignores failing tests by overriding `preCheck` instead of `checkPhase`. `preCheck` defines `CTEST_CUSTOM_TESTS_IGNORE` appropriately in `CTestCustom.cmake`.

JohnAZoidberg commented on pkgs/applications/science/misc/simgrid/default.nix:
```diff
@@ -18,13 +18,12 @@ in
-    repo = "simgrid";
-    rev = "v${replaceChars ["."] ["_"] version}";
-    sha256 = "1v0dwlww2wl56ms8lvg5zwffzbmz3sjzpkqc73f714mrc9g02bxs";
+  src = fetchgit {
```
There's `fetchFromGitLab`. You just need to supply `domain = "framagit.org"`.
That'll download the tarball directly without needing to fetch all the git stuff.

mpoquet replied:
Thanks, I did not know about `fetchFromGitlab`. Should be fixed in aa29b79e8a2.

mpoquet committed:
simgrid: fetchgit -> fetchFromGitLab

mpoquet committed:
simgrid: override checkPhase -> preCheck

This commit changes how failing tests are ignored.
- Previously, the whole checkPhase was overridden.
- Now, preCheck generates a ctest ignore file.

ctest doc: https://gitlab.kitware.com/cmake/community/wikis/doc/ctest/Testing-With-CTest#customizing-ctest

aanderse commented on pkgs/applications/science/misc/simgrid/default.nix:
```diff
@@ -17,14 +17,16 @@ let
-    sha256 = "1v0dwlww2wl56ms8lvg5zwffzbmz3sjzpkqc73f714mrc9g02bxs";
+  pname = "simgrid";
+  version = "3.22.2";
+  name = "${pname}-${version}";
```
This is not required as it will already be set if `pname` and `version` are supplied.

mpoquet replied:
Should be solved by 13aa449bab5, thanks.

mpoquet committed:
simgrid: remove unneeded 'name' definition

aanderse commented:
@GrahamcOfBorg build simgrid

aanderse commented:
@mpoquet Do you know if python bindings working in the current version? When I built this locally I see `-- Python libs not found. Turn pybind11 off.` but I notice that `python3` is included in the `nativeBuildInputs`.

mpoquet commented:
@aanderse SimGrid integration tests execute the [tesh](https://framagit.org/simgrid/simgrid/blob/v3.22.2/tools/tesh/tesh.py) SimGrid internal script, that runs a process and checks whether its output is the expected one. The Python dependency comes from tesh.

The python binding is experimental in SimGrid-3.22.2 — I'm part of the SimGrid dev team btw.
I tried to support the python binding in this package by adding the `pybind11` dependency ([patch](https://pastebin.com/FLJWNZTr)), but a pybind11 test seems to fail in this case ([log](https://pastebin.com/bcWyvnb7)).

Additionally, explicitly setting `-Denable_python=off` still issues a warning due to [a SimGrid CMake issue](https://framagit.org/simgrid/simgrid/issues/15), so it just replaces a warning by another :(.

aanderse commented:
@mpoquet sounds good, just wanted to make sure. Thanks!
